src/app/reexportr-src/reexportr.directive.ts

```diff
@@ -1,17 +1,39 @@
-import { Directive, Input, TemplateRef, ViewContainerRef } from "@angular/core";
+import {
+  Directive,
+  Input,
+  TemplateRef,
+  ViewContainerRef,
+  OnInit,
+  OnDestroy,
+  EmbeddedViewRef
+} from "@angular/core";
 
 @Directive({
   selector: "[reExportr]"
 })
-export class ReExportrDirective {
+export class ReExportrDirective implements OnInit, OnDestroy {
+  context = {
+    $implicit: undefined,
+    reExportr: undefined
+  };
+  embeddedView: EmbeddedViewRef<any>;
+
   constructor(private _tr: TemplateRef<any>, private _vcr: ViewContainerRef) {}
 
+  ngOnInit() {
+    this.embeddedView = this._vcr.createEmbeddedView(this._tr, this.context);
+  }
+
+  ngOnDestroy() {
+    this._vcr.clear();
+  }
```

Comment:
Is there any reason we need this?

See this stackblitz:
https://stackblitz.com/edit/angularnglet-old-x9cnyy

If you click "Destroy view", the inner component will log correctly that it's been destroyed as well.
So it seems that Angular is correctly taking over controlling this view, so I'm not sure we need it.

Reply:
I've done some memory benchmarks on the more extreme example and can't find any refernce to a memory leak. Memory seems to stay consistent on 1.2MB with my example with 10,000 paragraph tags after destroying and creating the elements many time. 

I think _I_ needed a lesson in trusting Angular to do some cleanup 😅  Good catch